config/import-control.xml

```diff
@@ -64,6 +64,15 @@
     <allow class="java.text.MessageFormat"/>
   </file>
 
+  <file name="AbstractAutomaticBean">
+    <allow class="java.beans.PropertyDescriptor" local-only="true"/>
+  </file>
+  <file name="Main">
+    <allow class="com.puppycrawl.tools.checkstyle.AbstractAutomaticBean.OutputStreamOptions"
+           local-only="true"/>
+    <allow pkg="javax.swing"/>
```

Comment:
Why do we need to allow this one?

Reply:
This file ruleset for `Main` that you commented on seems to override this rule below:
```
  <subpackage name="gui" strategyOnMismatch="disallowed">
    <allow class="com.puppycrawl.tools.checkstyle.xpath.XpathQueryGenerator" />


    <allow pkg="java.awt"/>
    <allow pkg="java.io"/>
    <allow pkg="javax.swing"/>
```

If I remove `<allow pkg="javax.swing"/>`(from row 73 where your comment is),this is the output: 

```
[INFO] --- maven-antrun-plugin:3.1.0:run (ant-phase-verify) @ checkstyle ---
[INFO] Executing tasks
[INFO]      [echo] Checkstyle started (checkstyle-checks.xml): 22/03/2023 03:47:54 PM
[INFO] [checkstyle] Running Checkstyle  on 1498 files
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/gui/Main.java:24:1: Disallowed import - javax.swing.SwingUtilities. [ImportControlMain]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/gui/Main.java:25:1: Disallowed import - javax.swing.WindowConstants. [ImportControlMain]
```

Reply:
Do you have any ideas on how to overcome that?

Reply:
@stoyanK7 can you specify full path to Main so that check doesn’t get confused?

Reply:
@nrmancuso It seems I can not.

![Screenshot from 2023-03-22 16-21-29](https://user-images.githubusercontent.com/23459549/226952736-4961121c-f83a-4d49-8d2e-62a233c13bd7.png)

```
$ mvn clean compile antrun:run@ant-phase-verify                                                                                                                                                 4:05:19 pm
[INFO] Scanning for projects...

...
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/Main.java:45:1: Disallowed import - com.puppycrawl.tools.checkstyle.AbstractAutomaticBean.OutputStreamOptions. [ImportControlMain]

```

Reply:
Done. Turns out the import control rule has to be at the end of the file.